Multi schema data block

SpencerBinXia opened the pull request.

SpencerBinXia committed:
initial commit for schemas data source

SpencerBinXia committed:
add optional pattern matching to data source

SpencerBinXia committed:
trim parameter names

SpencerBinXia committed:
data source passes smoke test

SpencerBinXia committed:
working pattern parameters

SpencerBinXia committed:
refactoring and add markup pages

SpencerBinXia committed:
refactor to meet TF OSS conventions

SpencerBinXia committed:
set data source schema resource ID including optional args and remove example

kaypeter87 reviewed (approved):
LGTM on first pass thru 👍 

I do not know this provider well so I'm deferring to you on the end-to-end test passing.

boekkooi-lengoo committed:
Use docker healthcheck (#168)

* Use docker healthcheck

Use a docker healthcheck with pg-isready instead of a custom bash script. It's just one thing less to maintain.

Also see:
- https://www.postgresql.org/docs/9.4/app-pg-isready.html
- https://docs.docker.com/engine/reference/builder/#healthcheck

* Set USER

This fixes the following error message.
```
Fix `FATAL:  role "root" does not exist`
```

Also see:
- https://docs.docker.com/engine/reference/builder/#user
- https://github.com/docker-library/postgres/blob/3bb4804/14/bullseye/Dockerfile#L23

* Use docker compose instead of docker-compose

* Update terraform SDK libs

Co-authored-by: Cyril Gaudin <cyril.gaudin@gmail.com>

boekkooi-lengoo committed:
Use docker healthcheck (#168)

* Use docker healthcheck

Use a docker healthcheck with pg-isready instead of a custom bash script. It's just one thing less to maintain.

Also see:
- https://www.postgresql.org/docs/9.4/app-pg-isready.html
- https://docs.docker.com/engine/reference/builder/#healthcheck

* Set USER

This fixes the following error message.
```
Fix `FATAL:  role "root" does not exist`
```

Also see:
- https://docs.docker.com/engine/reference/builder/#user
- https://github.com/docker-library/postgres/blob/3bb4804/14/bullseye/Dockerfile#L23

* Use docker compose instead of docker-compose

* Update terraform SDK libs

Co-authored-by: Cyril Gaudin <cyril.gaudin@gmail.com>

SpencerBinXia committed:
Acceptance for tests for schemas data source

SpencerBinXia committed:
add public reserved schema to comments

SpencerBinXia committed:
always include public schema regardless of include_system_schemas argument

SpencerBinXia committed:
change tests to account for variable temp pg_ schemas

SpencerBinXia committed:
Further clarifying comment for expected # of schemas in postgresql_schemas tests

SpencerBinXia committed:
change LIKE/NOT LIKE pattern matching options to take multiple values using ANY/ALL operators

boekkooi-lengoo committed:
Fix "Error: could not execute revoke query: pq: tuple concurrently updated" (#169)

* Show "tuple concurrently updated" error

The "tuple concurrently updated" error occurs when multiple connections do grants and revokes at the same time.

Also see
- https://www.postgresql.org/message-id/CAHyXU0y3M1Znv=MJ7u1y3pSMutALcoryQx9-v8c-WkaTdvvQ7Q@mail.gmail.com
- https://community.pivotal.io/s/article/Query-Fails-with-ERROR--tuple-concurrently-updated?language=en_US

* Restrict grant/revoke to one at a time

By using a mutex we ensure that we only execute a single GRANT or REVOKE at the same time fixing the error "Error: could not execute revoke query: pq: tuple concurrently updated"
which was caused by doing multiple grants and revokes at the same time.

* Update issues number

Now that the PR is created we have an issue number!

* Use pgLockRole instead of a mutex

By using the pgLockRole locking methods (which uses `pg_advisory_xact_lock`) we avoid locking to the provider and instead let postgres handle this.

boekkooi-lengoo committed:
Fix "Error: could not execute revoke query: pq: tuple concurrently updated" (#169)

* Show "tuple concurrently updated" error

The "tuple concurrently updated" error occurs when multiple connections do grants and revokes at the same time.

Also see
- https://www.postgresql.org/message-id/CAHyXU0y3M1Znv=MJ7u1y3pSMutALcoryQx9-v8c-WkaTdvvQ7Q@mail.gmail.com
- https://community.pivotal.io/s/article/Query-Fails-with-ERROR--tuple-concurrently-updated?language=en_US

* Restrict grant/revoke to one at a time

By using a mutex we ensure that we only execute a single GRANT or REVOKE at the same time fixing the error "Error: could not execute revoke query: pq: tuple concurrently updated"
which was caused by doing multiple grants and revokes at the same time.

* Update issues number

Now that the PR is created we have an issue number!

* Use pgLockRole instead of a mutex

By using the pgLockRole locking methods (which uses `pg_advisory_xact_lock`) we avoid locking to the provider and instead let postgres handle this.

Jell committed:
Add go mod vendor to linter, user correct version of go (#146)

Jell committed:
Add go mod vendor to linter, user correct version of go (#146)

Jell committed:
Add support for RDS IAM credentials (#134)

Jell committed:
Add support for RDS IAM credentials (#134)

alec-rabold committed:
postgresql_grant support for procedures and routines (#128)

* add postgresql_grant support for procedures/routines

* update postgresql_grant docs to add procedures/routines

* check version for procedures/routines in tests

* move compatibility test before creating objects

* fixup! Merge remote-tracking branch 'origin/master' into feature/grant-procedures-and-routines

Co-authored-by: Cyril Gaudin <cyril.gaudin@gmail.com>

alec-rabold committed:
postgresql_grant support for procedures and routines (#128)

* add postgresql_grant support for procedures/routines

* update postgresql_grant docs to add procedures/routines

* check version for procedures/routines in tests

* move compatibility test before creating objects

* fixup! Merge remote-tracking branch 'origin/master' into feature/grant-procedures-and-routines

Co-authored-by: Cyril Gaudin <cyril.gaudin@gmail.com>

cyrilgdn committed:
v1.15.0

cyrilgdn committed:
Fix CHANGELOG

SpencerBinXia committed:
change schemas output to TypeSet and adjust test cases to not rely on indexing

SpencerBinXia committed:
Merge remote-tracking branch 'origin/master' into multi-schema-data-block

SpencerBinXia committed:
remove logging, fix markdown page
